### DISCUSSION TITLE
How to utilize Docker Compose Profiles in Dev Container?

### DISCUSSION THREAD

**Discussion by crimsonvspurple** [devcontainers/spec#494] [Q&A]
I have a dev container system that I'm trying to use for CI too. For this, I feel the docker compose profile feature would simplify a lot of things but I'm not sure if I'm missing something simple...I can't find anything related to this.

How can I select a profile when building/running a dev container (using docker-compose) based on some environment variable such as CI=true?

Ref:
https://github.com/devcontainers/cli
https://github.com/devcontainers/ci
https://docs.docker.com/reference/compose-file/services/#profiles

**Comment by chrmarti** [comment]
This is currently not supported. See https://github.com/microsoft/vscode-remote-release/issues/5795 and https://github.com/devcontainers/cli/issues/669.

**Reply by AlekSi** [reply]
There is a PR from the community to implement this, PTAL.

**Reply by jenningsbrad73-cyber** [reply]
> There is a PR from the community to implement this, PTAL.

**Comment by SiegfredRodriguez** [comment]
Im just wondering, docker image already has runArgs, why not allow it on docker compose?

so we can just add the --profile parameter

**Comment by 1010101001010** [comment]
..
